src/account_manager/mod.rs

```diff
@@ -76,6 +76,44 @@ impl AccountManager {
         Ok(self.accounts.read().await.clone())
     }
 
+    /// Removes and returns the latest account (account with the largest account index).
+    pub async fn remove_latest_account(&mut self) -> crate::Result<Option<AccountHandle>> {
```

Comment:
I think this should return (), because if you still have the accounthandle and call a method on it which would store the account again, it wouldn't be removed